go.mod

```diff
@@ -17,7 +17,7 @@ module github.com/apache/trafficcontrol/v8
 // specific language governing permissions and limitations
 // under the License.
 
-go 1.23.0
+go 1.23.4
```

Comment:
Hmm, this will mean that a user cannot compile ATC using Go 1.23.3, and Go 1.23.4 or higher is required. Usually, minor versions are bugfixes and Go language changes are only introduced in major versions, so do we want to require specific minor versions?

Reply:
My reason is for aligning the Go version with .env and GO_VERSION is to maintain consistency across the development environment and avoid potential version mismatches.
I understand your perspective about not wanting to be overly restrictive with minor versions.
Would it better if we only watch & change the major/minor versions, and the patch starts keep 0?